esrc/rescue_success_mail.php

```diff
@@ -0,0 +1,172 @@
+<?php
+// Mark all entry pages with this definition. Includes need check check if this is defined
+// and stop processing if called direct for security reasons.
+if (!defined('ESRC')) define('ESRC', TRUE);
+
+require_once '../class/db.class.php';
+require_once '../class/rescue_mail.class.php';
+require_once '../class/users.class.php';
+require_once '../class/rescue.class.php';
+require_once '../class/systems.class.php';
+require_once '../class/output.class.php';
+
+// check if the user is alliance member
+if (!Users::isAllianceUserSession())
+{
+	// check if last login was already a non auth user
+	if (isset($_SESSION['AUTH_NOALLIANCE']))
+	{
+		// set redirect to root path
+		$_redirect_uri = Config::ROOT_PATH;
+	}
+	else
+	{
+		// set redirect to requested path
+		$_redirect_uri = $_SERVER['REQUEST_URI'];
+	}
+
+	// void the session entries on 'attack'
+	session_unset();
+	// save the redirect URL to current page
+	$_SESSION['auth_redirect']=$_redirect_uri;
+	// set a flag for alliance user failure
+	$_SESSION['AUTH_NOALLIANCE'] = 1;
+	// no, redirect to home page
+	header("Location: ".Config::ROOT_PATH."auth/login.php");
+	// stop processing
+	exit;
+}
+
+// prepare DB object
+$database = new Database();
+// create the query
+$reqID = (isset($_REQUEST['req'])) ? $_REQUEST['req'] : '';
+
+/* get all rescue information */
+$rescue = new Rescue($database);
+$systems = new Systems($database);
+$request = $rescue->getRequest($reqID);
+
+const EVEMAIL_DELIMITER = ';';
+
+class RescueSuccess {
+
+	public function generateTenderThanksMail($systemName, $sowPilot, $cacheTenders, $coordinator)
+	{
+		$subject = "Successful Rescue in " . $systemName;
+		$toList = $sowPilot . EVEMAIL_DELIMITER . $cacheTenders;
+
+		$bodyText = "Greetings, Signaleers!
+
+Multiple times a week pilots across New Eden are saved by rescue caches sown and tended by devoted Signaleers like yourselves.
+Often these rescues are accomplished so quickly that few people even know they happened.
+
+Recently a pilot was rescued using a cache in $systemName, sown by $sowPilot, which all of you contributed to keeping alive.
+
+Thank you for your dedication to the ESRC program. Your hard work has paid off!
+
+Yours in service,
+
+$coordinator,
+EvE-Scout Rescue
+https://evescoutrescue.com/
+";
```

Comment:
regarding the mail body.. see my note above with the SQL statement.

(adding the dates of each pilots last tend)